src/riak_kv_qry_worker.erl

```diff
@@ -262,7 +262,31 @@ throttling_spawn_index_fsms(State) ->
 estimate_query_size(#state{n_subqueries_done = NSubqueriesDone} = State)
   when NSubqueriesDone < 2  ->
     State;
-
```

Comment:
Even if it is received, the fact remains that we have exceeded the configured limit.

The simplicity of the size checks here is both the strength and the weakness. Ideally, we should be proactively monitoring the amount of memory used, by this query as well as by others served by us, and evaluate chances of successful delivery of the result to the client relative to that. We could, additionally, throttle the vnodes as a means to maintain safe levels of available memory. We should also keep track of the delivery times, to estimate our chances of getting the data in and out before the client request times out.

Instead, we shove responsibility to the user, and only promise the simple checks will be performed and limits observed as configured.